### PR TITLE
feat: added support for adding labels to default metrics

### PR DESCRIPTION
**Pull request by rkoval**
I believe this resolves https://github.com/siimon/prom-client/issues/224. I was also needing a mechanism like this for applying labels to worker processes when running in a clustered environment.

I did it a bit more generically than just the suggested `collectDefaultMetrics({includeWorkerId: true})` because our project setup prevents us from using lib/cluster.js. We're using a pm2 wrapper around our nodejs parent process which has some nuances when using the nodejs `cluster` module directly from lib/cluster.js. As such, this should appease a wider audience of users who just want to generically apply labels to default metrics (including myself)

### CODE REVIEW

File: test/defaultMetricsTest.js [hunk 1]
```diff
@@ -71,6 +71,21 @@ describe('collectDefaultMetrics', () => {
 		});
 	});
 
+	it('should apply labels to metrics when configured', () => {
+		const labels = {
```
Comment (SimenB) [inline test/defaultMetricsTest.js:75]: can you add a `expect.assertions(5)` (or whatever the number is)?

File: lib/metrics/version.js [hunk 2]
```diff
@@ -8,24 +8,25 @@ const NODE_VERSION_INFO = 'nodejs_version_info';
...
 module.exports = (registry, config = {}) => {
 	const namePrefix = config.prefix ? config.prefix : '';
+	const labels = config.labels ? config.labels : {};
+	const labelNames = Object.keys(labels);
 
 	const nodeVersionGauge = new Gauge({
 		name: namePrefix + NODE_VERSION_INFO,
 		help: 'Node.js version info.',
-		labelNames: ['version', 'major', 'minor', 'patch'],
+		labelNames: ['version', 'major', 'minor', 'patch'].concat(labelNames),
```
Comment (SimenB) [inline lib/metrics/version.js:17]:
```suggestion
		labelNames: ['version', 'major', 'minor', 'patch', ...labelNames],
```

I believe our node version support covers this syntax? Comment applies to the other `concat` calls as well

Comment (rkoval) [reply lib/metrics/version.js]: https://github.com/rkoval/prom-client/commit/c34d28d30997c44b8263f84b92edea9fb1c6e97d

File: test/defaultMetricsTest.js (hunk 1 above)
Comment (rkoval) [reply test/defaultMetricsTest.js:75]: https://github.com/rkoval/prom-client/commit/82a7106ec03f8b97c4ffd861d163d8b18e56ba6d

Comment (zbjornson) [reply test/defaultMetricsTest.js:75]: https://jestjs.io/docs/en/expect#expectassertionsnumber - it ensures that that many assertions are made, which makes sure tests don't falsely pass if some bug causes the async callback not to run.

File: test/defaultMetricsTest.js [hunk 3]
```diff
@@ -84,6 +84,7 @@ describe('collectDefaultMetrics', () => {
 				expect(metricValue.labels).toMatchObject(labels);
 			});
 		});
+		expect.assertions(45);
```
Comment (zbjornson) [inline test/defaultMetricsTest.js]: (I'm not sure if it matters, but usually this is the first line of the `it` function.)

**Comment by zbjornson** [comment]
I like this approach!

Sorry for the hassle but can you rebase on master please?

File: test/defaultMetricsTest.js (hunk 1 above)
Comment (rkoval) [reply test/defaultMetricsTest.js:73]: do you know why this function produces inconsistent results / doesn't always have the same number of assertions? i can actually reproduce the inconsistencies locally: 
image
the first run has 47 assertions, but the second only has 45 (i didn't change any code between each run)

**Comment by rkoval** [comment]
@zbjornson this has been rebased on the current master

File: test/defaultMetricsTest.js (hunk 1 above)
Comment (SimenB) [reply test/defaultMetricsTest.js:73]: probably something async we're not properly waiting for

Comment (rkoval) [reply test/defaultMetricsTest.js:73]: ok, what would you like to do here? again, this type of assertion isn't used anywhere else in the codebase, so i might just have to close this PR and just use our own fork if it'll be blocked without it. i unfortunately don't have time to debug the issue

File: README.md [hunk 4]
```diff
@@ -80,6 +80,16 @@ const prefix = 'my_application_';
 collectDefaultMetrics({ prefix });
 ```
 
+To apply generic labels to all default metrics, pass an object to the `labels` property (useful if you're working in a clustered environment):
+
+```js
+const client = require('prom-client');
+const collectDefaultMetrics = client.collectDefaultMetrics;
+collectDefaultMetrics({
+  labels: { NODE_APP_INSTANCE: process.env.NODE_APP_INSTANCE },
```
Comment (SimenB) [inline README.md:89]: need to add this to the TS types: https://github.com/siimon/prom-client/blob/48ed6c8b8112c5edf2b58985491a79632abe3112/index.d.ts#L567-L572

File: test/defaultMetricsTest.js (hunk 1 above)
Comment (SimenB) [reply test/defaultMetricsTest.js]: went with a slightly different approach

**Comment by SimenB** [comment]
@zbjornson @siimon PTAL

**Comment by siimon** [comment]
👍

**Pull request merged**